patterns/grok-patterns

```diff
@@ -35,7 +35,7 @@ HOSTPORT %{IPORHOST}:%{POSINT}
 
 # paths
 PATH (?:%{UNIXPATH}|%{WINPATH})
-UNIXPATH (/([\w_%!$@:.,+~-]+|\\.)*)+
+UNIXPATH (?>\/[[\w_%!$@:.,-]+|\\.]*)+
```

Comment:
Will try to explain what I did here:

> Two, it seems like the character class [....] was wrapped with another [...] and includes the +|\\. bits, which feels incorrect to me.

I changed the first level parenthesis group in favor of the brackets to there is only one "bigger" group, but looks in the process of cleaning this up I eat the `+~` characters that I understand are meaningful for unix file patterns.

Keeping the changes to:

* The atomic group.
* Only having one group.

Makes sense for you?